pytorch_lightning/trainer/data_loading.py

```diff
@@ -261,17 +261,23 @@ def _reset_eval_dataloader(
             mode: Either `'val'` or `'test'`
 
         Returns:
-            Tuple (num_batches, dataloaders)
+            Tuple (num_batches, dataloaders, ref_dataloaders)
         """
         # use the training loader as val and test when overfitting
         if self.overfit_batches > 0:
             dataloaders = self.request_dataloader(getattr(model, 'train_dataloader'))
+            # ref_dataloaders for comparing with train_dataloader
+            ref_dataloaders = self.request_dataloader(getattr(model, f'{mode}_dataloader'))
```

Comment:
this will cause the dataloaders to be called twice... this is a problem since users don't want this to happen often.

Reply:
Ahh, I see. I will try to find other way.